cluster/watchdog.go

```diff
@@ -121,6 +121,9 @@ func (w *Watchdog) rescheduleContainers(e *Engine) {
 			}
 		}
 
+		// Clear out the network configs, since we cannot create a
+		// container with multiple networks.
+		c.Config.NetworkingConfig.EndpointsConfig = nil
```

Comment:
Prior to this change, we essentially ignored the trouble; since we attempted to preserve the network configuration, if the container was connected to a bridge network, we'd try to recreate it on a new node with that same network config. If there was a network with the same name, it'd work; otherwise, it'd fail.

I can keep around networks that do not have the `swarm` or `global` scope if you'd like this change to have as little impact as possible.